### PR TITLE
Ticket651 add ioc cryosms

### PR DESCRIPTION
**Pull request by ChrisColeExpControl**
### Description of work

Added CRYOSMS IOC with basic functionality

### To test

https://github.com/ISISComputingGroup/IBEX/issues/4826

### Acceptance criteria

IOC starts correctly and passes the basic system tests.

---

#### Code Review

- [ ] A copy of the manual has been placed on the shared drive
- [ ] Pertitent information has been stored in the [wiki](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Specific-Device-IOC)
- [ ] Does the IOC conform to IBEX standards?
    - [PV naming](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/PV-Naming).
    - [Disable records](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Disable-records)
    - [Record simulation](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Record-Simulation)
    - [Finishing touches](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/IOC-Finishing-Touches)
- [ ] If an OPI has been modified, does it conform to the [style guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/OPI-Creation)?
- [ ] Do the IOC and support module conform to the new [build guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Reducing-Build-Dependencies)
- [ ] Have the changes been recorded appropriately in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?
- [ ] Is the device's flow control setting correct? [For most devices flow control should be OFF](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Flow-control).

### Functional Tests

- IOC responds correctly in:
    - [ ] Devsim mode
    - [ ] Recsim mode
    - [ ] Real device, if available
- [ ] Supplementary IOCs (`..._0n` where `n>1`) run correctly
- [ ] Log files do not report undefined macros (serach for `macLib: macro` to find instances of `macLib: macro [macro name] is undefined...`

### CODE REVIEW

File: CRYOSMS/configure/RELEASE [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+READASCII=$(SUPPORT)/ReadASCII/master
+SNCSEQ=$(SUPPORT)/seq/master
+SQLITE=$(SUPPORT)/sqlite/master
+SSCAN=$(SUPPORT)/sscan/master
+STREAMDEVICE=$(SUPPORT)/StreamDevice/master
+CRYOSMS=$(SUPPORT)/cryosms/master
+UTILITIES=$(SUPPORT)/utilities/master
+ZLIB=$(SUPPORT)/zlib/master
+GSL=$(SUPPORT)/gsl/master
+GSL_SUPPORT=$(GSL)/EPICS
```
Comment (DominicOram) [inline CRYOSMS/configure/RELEASE]: There are a lot of additional paths in here that I don't think are required. https://github.com/ISISComputingGroup/EPICS-base/blob/master/src/template/base/top/configure/RELEASE holds the list of the bare minimum ones that you need and I don't think you're using many of these others?

File: CRYOSMS/CRYOSMS-IOC-01App/src/build.mak [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+$(APPNAME)_LIBS += stream
+$(APPNAME)_LIBS += ReadASCII
+$(APPNAME)_LIBS += FileList
+$(APPNAME)_LIBS += calc sscan
+$(APPNAME)_LIBS += utilities
+$(APPNAME)_LIBS += asyn
+$(APPNAME)_LIBS += pcre libjson zlib
+$(APPNAME)_LIBS += efsw
+$(APPNAME)_LIBS += asubFunctions gsl gslSupport
+$(APPNAME)_LIBS += cryosms
```
Comment (DominicOram) [inline CRYOSMS/CRYOSMS-IOC-01App/src/build.mak]: There are a lot of additional dbds and libs in here that I don't think are required. https://github.com/ISISComputingGroup/EPICS-base/blob/master/src/template/base/top/iocApp/src/build.mak holds the list of the bare minimum ones that you need and I don't think you're using many of these others?

File: CRYOSMS/CRYOSMS-IOC-02App/src/Makefile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+TOP=../..
+# This file should do very little - it's purpose is to set the APPNAME and then load build.mak
+
+# this definition is used in build.mak
+APPNAME=CRYOSMS-IOC-01
```
Comment (DominicOram) [inline CRYOSMS/CRYOSMS-IOC-02App/src/Makefile]: This should be `CRYOSMS-IOC-02`

File: CRYOSMS/iocBoot/iocCRYOSMS-IOC-01/st-common.cmd [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+epicsEnvSet "STREAM_PROTOCOL_PATH" "$(CRYOSMS)/data"
+
+##ISIS## Load common DB records 
+< $(IOCSTARTUP)/dbload.cmd
+
+############################
+## Load our record instances
+############################
+dbLoadRecords("$(CRYOSMS)/db/cryosms.db","P=$(MYPVPREFIX)$(IOCNAME):,PORT=$(PORT=STREAM_PORT),ASYN_PORT=ASYN_PORT,RECSIM=$(RECSIM=0),DISABLE=$(DISABLE=0),PROTO=$(PROTO=cryosms.proto)")
```
Comment (Tom-Willemsen) [inline CRYOSMS/iocBoot/iocCRYOSMS-IOC-01/st-common.cmd]: Why do we need to pass `PROTO` as an argument here, it does not look like it is used by the db?

**Pull request merged**